packages/ui/src/hooks/use-api.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+import { HTTPError } from 'ky';
+import { RequestErrorBody } from '@logto/schemas';
+
+function useApi<T extends any[], U>(api: (...arguments_: T) => Promise<U>) {
```

Comment:
`U` should be more specific i think.. like rename to `Response` and return something like `Promise<UseApi<Response>>` where 

```ts
type UseApi<T> = {
  result: T,
  fetch: ...,
  ...
}
```

Reply:
Just make sure I understand correctly:
 The ReturnType should be  sth like UseApi<Args, Response>, and hook func itself does not return a promise

Reply:
yes